### DISCUSSION TITLE
fsx.delete(string) should not recursively delete a directory

### DISCUSSION THREAD

**Discussion by jhulford** [humanwhocodes/humanfs#22] [Feedback]
Read your "Introducing fsx" post and I think the project you're laying out looks great, one major thing stood out to me in the post as concerning and that's recursively deleting a directory via the fsx.delete function.

While I think having a simple function to do the equivalent of `rmdir -r` is great and useful, abstracting it behind the same function that also deletes a single file is an invitation to dev to shoot themselves in the foot (or the head if this function is pointed at the wrong directory). All it would take is a minor bug in string concatenation and you've potentially lost a lot of valuable data.

I would suggest at least splitting the delete function into deleteFile and deleteDirectory, which would effectively force the developer to relay explicit intent to do one or the other instead of leaving the choice up to the path passed into the function - each implementation can then take the necessary precautions to ensure the path is a file or a directory.

**Comment by nzakas** [comment]
I agree. See https://github.com/humanwhocodes/fsx/discussions/17
